### PR TITLE
Fix single-tab window tear-off crash

### PR DESCRIPTION
**Pull request by tusharsnx**
## Validation Steps Performed
- Opened multi-tab terminal window with Narrator. Narrator can read characters from the tabs.
- Started a drag and drop (tear-off) of a tab, and it didn't crash. This was repeated multiple times.

### CODE REVIEW

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 1]
```diff
@@ -1051,7 +1051,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         // MSFT 33353327: We're purposefully not using _initializedTerminal to ensure we're fully initialized.
         // Doing so makes us return nullptr when XAML requests an automation peer.
         // Instead, we need to give XAML an automation peer, then fix it later.
-        if (!_IsClosing())
+        if (!_IsClosing() && !_detached)
```
Comment (lhecker) [inline src/cascadia/TerminalControl/TermControl.cpp:1054]: Does it make sense to instead check for `!_interactivity` since it's used below?

**Review by lhecker — APPROVED** [review]

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 1 above)
Comment (tusharsnx) [reply src/cascadia/TerminalControl/TermControl.cpp:1054]: When termcontrol isn't detached or closing we expect `_interactivity` to be non-null, no?

Given #10971 has happened in the past, I think `THROW_HR_IF_NULL(E_UNEXPECTED, _interactivity)` is better suited.

**Review by zadjii-msft — APPROVED** [review]
image

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 1 above)
Comment (DHowett) [reply src/cascadia/TerminalControl/TermControl.cpp:1054]: I'm OK with this for now

**Review by DHowett — APPROVED** [review]

**Pull request merged**